include/drivers/gpio.h

```diff
@@ -87,37 +400,42 @@ struct gpio_callback {
 };
 
 /**
- * @cond INTERNAL_HIDDEN
+ * @brief Pins associated with a port.
  *
- * GPIO driver API definition and system call entry points
- *
- * (Internal use only.)
- */
-typedef int (*gpio_config_t)(struct device *port, int access_op,
-			     u32_t pin, int flags);
-typedef int (*gpio_write_t)(struct device *port, int access_op,
-			    u32_t pin, u32_t value);
-typedef int (*gpio_read_t)(struct device *port, int access_op,
-			   u32_t pin, u32_t *value);
-typedef int (*gpio_manage_callback_t)(struct device *port,
-				      struct gpio_callback *callback,
-				      bool set);
-typedef int (*gpio_enable_callback_t)(struct device *port,
-				      int access_op,
-				      u32_t pin);
-typedef int (*gpio_disable_callback_t)(struct device *port,
-				       int access_op,
-				       u32_t pin);
-typedef u32_t (*gpio_api_get_pending_int)(struct device *dev);
+ * Value of this type identifies pins, or their value, associated with a port.
+ * Pin with index n, or its value, is identified by bit n (1U << n) in the value
+ * of type gpio_pins_t.
+ */
+typedef u32_t gpio_pins_t;
```

Comment:
This typedef probably doesn't belong in this file as it is architecture specific. I'm not sure where to put it though.